File: tools/bespoke_osis2mod/sword-1.9.0/bindings/objc/test/SwordManagerTest.m
//
//  SwordManagerTest.m
//  ObjCSword
//
//  Created by Manfred Bergmann on 02.03.16.
//
//

#import <XCTest/XCTest.h>
#import <ObjCSword/ObjCSword.h>

@interface SwordManagerTest : XCTestCase

@end

@implementation SwordManagerTest {
    SwordManager *mgr;
}

- (void)setUp {
    [super setUp];

    [Configuration configWithImpl:[[OSXConfiguration alloc] init]];
    [[FilterProviderFactory providerFactory] initWithImpl:[[DefaultFilterProvider alloc] init]];
    
    NSString *modulePath = [[[NSBundle bundleForClass:[self class]] resourcePath] stringByAppendingPathComponent:@"TestModules"];
    
    mgr = [SwordManager managerWithPath:modulePath];
}

- (void)testAvailableModules {
    XCTAssert(mgr != nil);
    
    NSInteger count = [[mgr allModules] count];
    XCTAssert(count > 0);
    NSLog(@"Modules: %lu", count);
}

- (void)testGetModule {
    SwordModule *mod = [mgr moduleWithName:@"KJV"];
    XCTAssert(mod != nil);
    XCTAssert([[mod name] isEqualToString:@"KJV"]);
}

- (void)testReloadWithRenderedKey_String {
    SwordModule *mod = [mgr moduleWithName:@"KJV"];

    [mod setKeyString:@"Gen 1:1"];
    NSString *text = [mod renderedText];
    XCTAssert(text != nil);
    XCTAssert(text.length > 0);
    NSLog(@"text: %@", text);
    
    [mgr reloadManager];
    mod = [mgr moduleWithName:@"KJV"];
    XCTAssert(mod != nil);
    XCTAssert([[mod name] isEqualToString:@"KJV"]);
}

- (void)testReloadWithRenderedKey_SwordKey {
    SwordModule *mod = [mgr moduleWithName:@"KJV"];
    
    SwordKey *key = [SwordKey swordKeyWithRef:@"Gen 1"];
    [mod setSwordKey:key];
    NSString *text = [mod renderedText];
    XCTAssert(text != nil);
    XCTAssert(text.length > 0);
    NSLog(@"text: %@", text);
    
    [mgr reloadManager];
    mod = [mgr moduleWithName:@"KJV"];
    XCTAssert(mod != nil);
    XCTAssert([[mod name] isEqualToString:@"KJV"]);
}

- (void)testReloadWithRenderedKey_CustomRender {
    SwordModule *mod = [mgr moduleWithName:@"KJV"];

    NSArray *textEntries = [(SwordBible *)mod renderedTextEntriesForRef:@"Gen 1" context:0];
    XCTAssert(textEntries != nil);
    NSLog(@"Entries: %lu", [textEntries count]);
    XCTAssert([textEntries count] == 31);
    
    [mgr reloadManager];
    mod = [mgr moduleWithName:@"KJV"];
    XCTAssert(mod != nil);
    XCTAssert([[mod name] isEqualToString:@"KJV"]);
}

@end

File: tools/bespoke_osis2mod/sword-1.9.0/bindings/objc/test/SwordManagerTest.swift
//
//  SwordManagerTest.swift
//  ObjCSword
//
//  Created by Manfred Bergmann on 02.03.16.
//
//

import XCTest

class SwordManagerTest: XCTestCase {

    var mgr: SwordManager?

    override func setUp() {
        super.setUp()
        
        let modulesDir = NSBundle(forClass:self.dynamicType).resourcePath!.stringByAppendingString("/TestModules")
        NSLog("modulesDir: \(modulesDir)")
        
        Configuration.configWithImpl(OSXConfiguration())
        FilterProviderFactory().initWithImpl(DefaultFilterProvider())
        mgr = SwordManager(path:modulesDir)
    }
    
    func testAvailableModules() {
        XCTAssert(mgr != nil)
        XCTAssert(mgr?.modules.count > 0)
        NSLog("modules: \(mgr?.modules.count)")
    }
    
    func testGetModule() {
        let mod = mgr?.moduleWithName("kjv")
        XCTAssert(mod != nil)
        XCTAssert(mod?.name() == "KJV")
    }

    func testReload() {
        var mod = mgr?.moduleWithName("kjv")
        
        mgr?.reloadManager()
        mod = mgr?.moduleWithName("kjv")
        
        XCTAssert(mod != nil)
        XCTAssert(mod?.name() == "KJV")
    }

    func testReloadWithKeyString() {
        var mod = mgr?.moduleWithName("kjv")
        
        let te = mod?.renderedTextEntriesForRef("Gen 1")
        XCTAssert(te?.count > 0)
        NSLog(te![0] as! String)
        
//        mod?.setKeyString("Gen 1")
//        let text = mod?.renderedText()
//        XCTAssert(text != nil)
//        XCTAssert(text?.lengthOfBytesUsingEncoding(NSUTF8StringEncoding) > 0)
//        NSLog(text!)
        
        mgr?.reloadManager()
        mod = mgr?.moduleWithName("kjv")
        
        XCTAssert(mod != nil)
        XCTAssert(mod?.name() == "KJV")
    }
}
